lib/client/client-spec.js

```diff
@@ -0,0 +1,79 @@
+var mediator = require('fh-wfm-mediator/lib/mediator');
+var sinon = require('sinon');
+var assert = require('assert');
+
+describe("Raincatcher Sync Client Initialisation", function() {
+
+  var mockSyncOptions = {
+    "sync_frequency": 5,
+    "storage_strategy": "dom",
+    "do_console_log": false
+  };
+
+  var mockDataSetId = "mockdatasetid";
+
+  var mockDataSetFilterParams = {filterParam1: "filtervalue"};
+  var mockDataSetMetadataParams = {metadataParam1: "metadatavalue"};
+
+  before(function() {
+    var self = this;
+
+    //Clearing the require cache to re-require the client
+    self.syncClient = require('../client');
+
+    self.mock$fh = {
+      sync: {
+        init: sinon.spy(),
+        notify: sinon.spy(),
+        manage: sinon.stub().callsArg(4)
+      }
+    };
+
+  });
+
+
+  it("should not allow a manager to be created if sync has not been initialised", function(done) {
```

Comment:
No need for done(). Just return the promise. (This applies for most, if not all, of the async tests in this PR.)